src/pngimage.cpp

```diff
@@ -589,9 +589,9 @@ namespace Exiv2 {
                     ExifParser::encode(blob, littleEndian, exifData_);
                     if (blob.size() > 0)
                     {
-                        static const char exifHeader[] = { 0x45, 0x78, 0x69, 0x66, 0x00, 0x00 };
-                        std::string rawExif =   std::string(exifHeader, 6)
-                                              + std::string((const char*)&blob[0], blob.size());
+                        const std::string exifHeader{"Exif\0\0", 6};
+                        std::string rawExif = exifHeader
+                                              + std::string((const char*)blob.data(), blob.size());
```

Comment:
suggestion: drop the C style cast while we're at it.

Reply:
Hm, could do. But I did not touch the cast at all and there are plenty more of them in the lib.